spec/shakapacker/test_app/package.json

```
@@ -9,5 +9,6 @@
   },
   "devDependencies": {
     "right-pad": "^1.0.1"
-  }
+  },
+  "packageManager": "yarn@1.22.24"
```

We need the same in `spec/dummy/package.json`. Lack of this is causing failures there. Wonder if this lacks some better handling.

To repro dummy fail:
```
- remove `spec/dummy/public/packs-test` directory
- run bundle exec rake run_spec:dummy
```

This should fail. Trying to run `rake assets:precompile` in the dummy dir:
```
rake aborted!
Shakapacker::Utils::Manager::Error: You have not got "packageManager" set in your package.json meaning that Shakapacker will use npm
but you've got a yarn.lock file meaning you probably want
to be using yarn instead.

To make this happen, set "packageManager" in your package.json to yarn@1.22.19
/Users/tom.dracz/projects/shakapacker/lib/shakapacker/utils/manager.rb:25:in `error_unless_package_manager_is_obvious!'
/Users/tom.dracz/projects/shakapacker/lib/tasks/shakapacker/check_manager.rake:7:in `block (2 levels) in <main>'
```

So the compile on the fly probably also fails but the issue is non-obvious. Should check how it behaves in real-life scenario to see if that's anything to patch 🤔

---

it does look like the version mismatch warning/error gets outputted correctly, which I notice is called in `railtie.rb` - I'm not super familiar with how railties actually work but should we actually be doing this check there instead?

---

oh hey yeah test output is going to `spec/dummy/test.log`, which has all the stuff - maybe the bug is that something isn't checking an exit code? fwiw I've tested this on some of my apps which use `react-rails` and they error fine.

---

interestingly setting `config.action_dispatch.show_exceptions = :rescuable` (which is the value we use at Ackama but currently `false` in `spec/dummy`) in `config/environments/test.rb` results in the tests passing - is it possible we're hitting an error that isn't actually an error..?

either way @tomdracz I've pushed up switching to using `railtie` to do the check - let me know if you think that's the right move, and we can pick up from there

---

Thanks, I'll play around. With this particular thing I presume it just might be React on Rails swallowing error and raising another one

---

Looks like at least we're erroring out early so nice improvement. https://github.com/shakacode/shakapacker/pull/430/commits/3a566ef32a1ef2c3d2117a657f4558e0ab361ac5 to patch the dummy spec failure itself